kernel_gateway/mixins.py

```diff
@@ -54,8 +55,14 @@ def prepare(self):
         """
         server_token = self.settings.get('kg_auth_token')
         if server_token:
-            client_token = self.request.headers.get('Authorization')
-            if client_token != 'token %s' % server_token:
+            client_token = self.get_argument('token', '')
+            if client_token == '':
+                client_token = self.request.headers.get('Authorization')
+                if client_token and client_token.startswith('token '):
+                    client_token = client_token[len('token '):]
```

Comment:
`len('token ')` should probably be computed once and stored a global in this module to avoid recompute it on every request.